spec/controllers/assignments_controller_spec.rb

```diff
@@ -59,26 +63,27 @@
     end
     it 'redirects to the index' do
       submit
-      expect(response).to redirect_to assignments_url
+      expect(response).to redirect_to roster_assignments_url
     end
   end
 
   describe 'GET #edit' do
     before :each do
       @assignment = create :assignment
+      @roster = create :roster
       when_current_user_is :whoever
     end
     let :submit do
-      get :edit, id: @assignment.id
+      get :edit, roster_id: @roster.id, id: @assignment.id
     end
     it 'finds the correct assignment' do
       submit
       expect(assigns.fetch :assignment).to eql @assignment
     end
     it 'populates a users variable of all users' do
-      user_1 = create :user
-      user_2 = create :user
-      user_3 = create :user
+      user_1 = create :user, rosters: [@roster] 
```

Comment:
IMHO there's something to be said for WET tests cases. There's something to be said for total isolation for each example, even if comes at the expense of repetitive LOC.

That being said, if a macro makes sense that that's a way to go here. I think the danger there is that said macro can (and almost always will) grow moving more and more setup logic out of the individual examples making it harder to understand (in the future) why tests fail.